Edit Issue

profornnan opened the pull request:
- 이슈 수정 API를 구현했습니다.
- transaction을 사용해서 Issue에 대한 정보뿐만 아니라 Label과 Assignee 정보까지 모두 update에 성공했을 때 COMMIT하고 실패했다면 ROLLBACK 하도록 했습니다.

resolved: #96

profornnan committed:
[Feat] Validate Edit Issue Input

Add validateEditIssueInput middleware

profornnan committed:
[Feat] EditIssue Model

Add model for editIssue

profornnan committed:
[Feat] EditIssue Service

Add service for editIssue

profornnan committed:
[Feat] EditIssue Controller

Add controller for editIssue

profornnan committed:
[Feat] EditIssue Route

Add route for editIssue

gyim1345 commented on back/src/middleware/validateInputs.js:
```diff
@@ -5,6 +5,22 @@ const validateCreateIssueInput = (req, res, next) => {
+      statusOpenClosed === true ||
+      statusOpenClosed === false ||
+      statusOpenClosed === undefined
+    )
```
따로 함수로 빼시는것도 괜찮을것 같습니다!

gyim1345 reviewed (approved):
수고하셨습니다!

rockpell reviewed (approved):
트랜잭션 적용하신거 좋네요
수고하셨습니다.

pieisland reviewed (approved):
고생하셨어요~ 디테일하게 생각 많이 하고 짜시는 것 같아요! 저도 그래야하는데 항상 놓치는 부분이 있는 것 같네요 ㅠㅠ.

profornnan committed:
Merge remote-tracking branch 'upstream/develop' into editIssue

profornnan committed:
[Feat] Validate Datatypes

Define validateStringOrUndefined function
Define validateBooleanOrUndefined function

profornnan committed:
[Refactor] validateEditIssueInput

Use validateDatatypes in validateEditIssueInput
